### PR TITLE
Update config.rst

### PR DESCRIPTION
**Pull request by anton-seaice**
Following on from https://github.com/payu-org/payu/issues/487#issuecomment-2297680663

Hopefully this clarifies how userscripts work (it will for me anyway :-) )

### CODE REVIEW

**Review by aidanheerdegen — DISMISSED** [review]
It's a good change I think. Thanks @anton-seaice 

I think we should put something in there about error handling to make it clear to users that a script or command that returns an error state can (and probably should) stop the model from running.

Now this doesn't have to happen in this PR, but it might make sense to do so.

File: docs/source/config.rst [hunk 1]
```diff
@@ -351,22 +351,18 @@ FMS based model only options:
...
-   Controls whether or not a collation job is submitted after model execution.
-
-   This is typically ``True``, although individual model drivers will often set
-   the default value to ``False`` if collation is unnecessary.
-
-   See above for specific ``collate`` options.
-
 ``userscripts``
    Namelist to include separate userscripts or subcommands at various stages of
    a payu submission. Inputs can be either script names (``some_script.sh``) or
```
[the remark this answers is not in the document]
Comment (anton-seaice) [reply docs/source/config.rst:359]: Yes I had that thought but didn't have a good replacement word!

File: docs/source/config.rst [hunk 2]
```diff
@@ -351,22 +351,18 @@ FMS based model only options:
...
    individual subcommands (``echo "some_data" > input.nml``, ``qsub
-   some_script.sh``).
+   some_script.sh``). Userscripts are run within the same PBS job as the model 
+   execution unless the script starts a new PBS job. Userscripts therefore have
+   the same compute, storage and network access as the model. The exception to 
+   this is when `payu setup` is called directly, then the relevant userscripts 
+   will run on the login node.
 
    Specific scripts are defined below:
```
[the remark this answers is not in the document]
Comment (anton-seaice) [reply docs/source/config.rst:367]: These some like more general descriptions than just user scripts ? Is this the best spot in the docs for them? However, I also can't see anywhere in the docs they obviously fit better ...

[the remark this answers is not in the document]
Comment (aidanheerdegen) [reply docs/source/config.rst:367]: It's a good point. If I can get back to it, probably worth making a section describing these steps better and referring back to that here.

File: docs/source/config.rst [hunk 3]
```diff
@@ -351,22 +351,18 @@ FMS based model only options:
...
-   Namelist to include separate userscripts or subcommands at various stages of
+   Configure userscripts or subcommands to run at various stages of
    a payu submission. Inputs can be either script names (``some_script.sh``) or
    individual subcommands (``echo "some_data" > input.nml``, ``qsub
-   some_script.sh``).
+   some_script.sh``). Userscripts are run within the same PBS job as the model 
+   execution unless the script starts a new PBS job. Userscripts therefore have
+   the same compute, storage and network access as the model. The exception to 
+   this is when `payu setup` is called directly, then the relevant userscripts 
+   will run on the login node.
```
Comment (jo-basevi) [inline docs/source/config.rst:365]: Another exception is the `sync` user scripts which runs on the `sync` pbs job. It's noted under the `sync` user script sub-section. Maybe the sync userscript section should be extended it make it a little more clear? e.g.:

""User-defined command to be called at the start of the `sync` PBS job. This is useful for any post-processing before syncing files to a remote archive. Note these scripts are only run within `sync`  PBS job so will not run unless automatic syncing is enabled or if `payu sync` is run manually.""

**Review by jo-basevi — APPROVED** [review]
Thanks for fixing up the documentation!

**Comment by anton-seaice** [comment]
Sorry I did the old trick of writing comments and then not clicking submit! ... See https://github.com/payu-org/payu/pull/489#discussion_r1728220872

File: docs/source/config.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (anton-seaice) [reply docs/source/config.rst]: ping @aidanheerdegen - Could the describtions go on the `Usage` page ?

[the remark this answers is not in the document]
Comment (aidanheerdegen) [reply docs/source/config.rst]: I think they naturally go in the design section

https://payu.readthedocs.io/en/stable/design.html

but happy to defer, not dying in a ditch about it.

**Comment by anton-seaice** [comment]
I think this is ready to go now :)

**Review by jo-basevi — APPROVED** [review]
Had a read through, and it looks good to me!

**Comment by anton-seaice** [comment]
Thanks @jo-basevi !

**Pull request merged**